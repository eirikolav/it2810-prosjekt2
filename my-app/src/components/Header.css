.grid-container {
    display: grid;
    grid-column-gap: auto;
    grid-template-rows: 70px 200px;
    grid-template-columns: auto; 
    margin-bottom: 10px;
}



.Title {
    display: grid;
    text-align: center;
    font-family: Brush Script MT, Brush Script Std, cursive;
    font-size: 40px;
    height: 500px;
    margin-top: 50px;
}


.Navbar {
    position: fixed;
    top: 0;
    max-height: 70px;
    width: 100vw;
    display: grid; /*generates block level grid*/
    grid-template-areas: "logo nav";
    box-shadow: 0 4px 8px 0 #00000033;
    z-index: 10;
}

.Logo {
    grid-area: logo;
    height: 70px;
    transition: 1s;
}

.Logo:hover {
    transform: scale(1.2);
    cursor: pointer;
}

.Nav {
    grid-area: nav;
    display: grid;
    grid-template-columns: repeat(4, auto);
    align-items: center;
    justify-items: center;
}

.theme {
    padding: 10px;
    outline: none;
    border: none;
    font-size: 20px;
    margin-left: 10px;
    color: white;
    font-weight: 600;
    background-color: white;
    border-radius: 10px;
    cursor: pointer;
    transition: 100ms;
}

#love {
    background-color: pink;
}

#whale {
    background-color: lightblue;
}

#sun {
    background-color: lightyellow;
}

.theme:hover {
    transform: scale(1.4);
}

.save {
    padding: 10px;
    outline: none;
    border: none;
    font-size: 20px;
    margin-left: 10px;
    color: white;
    font-weight: 600;
    background-color: rgba(33, 233, 27, 0.5);
    box-shadow: 0px 5px 0px 0px rgba(33, 233, 27, 0.5);
    border-radius: 10px;
    cursor: pointer;
    transition: 70ms;
}

.save:hover {
    background-color: rgba(247, 9, 227, 0.5);
    box-shadow: 0px 5px 0px 0px rgba(247, 9, 227, 0.5);
}

.save:active {
    transform: translateY(5px);
}

.Lion {
    display: none;
    grid-area: lion;
    margin: 0 20px 0 0;
    padding: 0;
    justify-self: end;
    font-size: 40px;
    border: none;
    background: none;
    outline: none;
    transition: 0.1s;
}

.Lion:active {
    transform: scale(1.2);
}


@media (max-width: 700px) {
    .Navbar {
        grid-template-areas: "logo lion" "nav nav";
    }
    .Nav  {
        grid-template-rows: repeat(4, auto);
        grid-template-columns: none;
        grid-row-gap: 20px;

        padding: 30px 0 30px;
        background-color:ghostwhite;
        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
        border-bottom-left-radius: 10px;
        border-bottom-right-radius: 10px;
    
    }
    .Lion {
        display: block;
    }
}