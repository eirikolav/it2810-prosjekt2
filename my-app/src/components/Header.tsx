import React, { useState, useEffect } from 'react';
import './Header.css';


export default function Header(): React.ReactElement {

    const [isNavVisible, setIsNavVisible] = useState(true);
    const [isSmallScreen, setIsSmallScreen] = useState(false);

    useEffect(() => {
        const mediaQuery = window.matchMedia("(max-width: 700px)");
        mediaQuery.addListener(handleMediaQueryChange);
        handleMediaQueryChange(mediaQuery);

        return () => {
            mediaQuery.removeListener(handleMediaQueryChange);
        };
    }, []);


    const handleMediaQueryChange = (mediaQuery : any) => {
        if (mediaQuery.matches) {
            setIsSmallScreen(true);
        } else {
            setIsSmallScreen(false);
        }
    };

    const toggleNav = () => {
        setIsNavVisible(!isNavVisible);
    };

    
    return (
        <div className="grid-container">
            <div className="Navbar">
                <img src={require("../pictures/logo192.png")} className="Logo" alt="logo"/>
                {(!isSmallScreen || isNavVisible) && (
                <nav className="Nav">
                    <a href="#">&#128150;</a>
                    <a href="#">&#128011;</a>
                    <a href="#">&#128526;</a>
                    <button>Lagre</button>
                </nav>
                )}
                <button onClick={toggleNav} className="Lion">&#129409;</button>
            </div>
            <div className="Title">
                <h1>Kul utstilling bro</h1>
            </div>
        </div>
    )
}






