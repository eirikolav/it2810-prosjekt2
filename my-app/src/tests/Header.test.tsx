import React from 'react'
import renderer from 'react-test-renderer'
import Header from "../components/Header"

test("Test Header", () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
})