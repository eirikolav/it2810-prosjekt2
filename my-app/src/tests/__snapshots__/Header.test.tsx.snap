// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header 1`] = `
<div
  className="grid-container"
>
  <div
    className="Navbar"
  >
    <img
      alt="logo"
      className="Logo"
      src="logo192.png"
    />
    <nav
      className="Nav"
    >
      <button
        className="theme"
        id="love"
      >
        💖
      </button>
      <button
        className="theme"
        id="whale"
      >
        🐋
      </button>
      <button
        className="theme"
        id="sun"
      >
        😎
      </button>
      <button
        className="save"
      >
        Lagre
      </button>
    </nav>
    <button
      className="Lion"
      onClick={[Function]}
    >
      🦁
    </button>
  </div>
  <div
    className="Title"
  >
    <h1 />
  </div>
</div>
`;
